components/ConnectToDiscordModal.tsx

```diff
@@ -1,23 +1,47 @@
-import React from 'react'
+import React, { useEffect, useState } from 'react'
 import { ModalCard, ModalSize } from './ModalCard'
 import NavLink from './NavLink'
-import LogoutContainer from './LogoutContainer'
 
 type ConnectToDiscordModalProps = {
   show: boolean
+  close: Function
 }
 
 const discordConnectPage = `${process.env.NEXT_PUBLIC_DISCORD_CALLBACK_URI}`
 
+const CONNECT_TO_DISCORD_LOCALSTORAGE_KEY = 'connect-to-discord'
+const SEVEN_DAYS = 1000 * 60 * 60 * 24 * 7
```

Comment:
Looks like there is another place that uses the `7 day` constant here:
https://github.com/garageScript/c0d3-app/blob/b1a6736084ac6cb1fde2a2d6fb57c2d64537979f/helpers/middleware/session.ts#L6

Can you make the `7 days` variable to be used as a constant?

I think there can probably be some little more refactoring to use constants to reduce duplicated code too:
https://github.com/garageScript/c0d3-app/search?q=1000+*+60+*+60+*+24+*+7

^ The refactoring should be another PR though. But I think you can start in this PR by making a constant for `7 days` to be used for any files in the future(`including this file`)

Reply:
I just looked through the codebase and I don't see any kind of constants file. So I think you'd be making the first one in this repo.

I'm thinking it would be best to have a `constants` folder in the root of this repo, and inside there can be individual files relating to some topic. For example we could have a `requestMethods.ts` file that would only contain constants used for requests, like `GET` or `POST`.

Thoughts?
@songz @hwong0305 @kondanna @tomrule007

Reply:
Sure, I'm for it, if it's an accepted practice.

Reply:
let's revisit this on another PR

Reply:
I think it's a good idea to store all our commonly used constants or variables that keeps getting reused into a constant folder. It would have us having to repeat ourselves and in the future if we need to change something, we can change it in one place instead of many different places.

Reply:
You want this refactor in this PR or a separate one?

Reply:
Decided to refactor in another PR: 
https://github.com/garageScript/c0d3-app/issues/1185